chemprop/utils.py

```diff
@@ -528,3 +528,86 @@ def save_smiles_splits(data_path: str,
+    if train_args.number_of_molecules != predict_args.number_of_molecules:
+        raise ValueError('A different number of molecules was used in training '
+                        f'model than is specified for prediction, {train_args.number_of_molecules} '
+                         'smiles fields must be provided')
+
+    # If atom-descriptors were used during training, they must be used when predicting and vice-versa
+    if train_args.atom_descriptors != predict_args.atom_descriptors:
+        raise ValueError('The use of atom descriptors is inconsistent between training and prediction. If atom descriptors '
+                         ' were used during training, they must be specified again during prediction using the same type of '
+                         ' descriptors as before. If they were not used during training, they cannot be specified during prediction.')
+
+    # If bond features were used during training, they must be used when predicting and vice-versa
+    if (train_args.bond_features_path is None) != (predict_args.bond_features_path is None):
+        raise ValueError('The use of bond descriptors is different between training and prediction. If you used bond'
+                         'descriptors for training, please specify a path to new bond descriptors for prediction.')
+
+    # if atom or bond features were scaled, the same must be done during prediction
+    if train_args.features_scaling != predict_args.features_scaling:
+        raise ValueError('If scaling of the additional features was done during training, the'
+                         'same must be done during prediction.')
+
+    # If atom descriptors were used during training, they must be used when predicting and vice-versa
+    if train_args.atom_descriptors != predict_args.atom_descriptors:
+        raise ValueError('The use of atom descriptors is inconsistent between training and prediction. '
+                         'If atom descriptors were used during training, they must be specified again '
+                         'during prediction using the same type of descriptors as before. '
+                         'If they were not used during training, they cannot be specified during prediction.')
+
+    # If bond features were used during training, they must be used when predicting and vice-versa
+    if (train_args.bond_features_path is None) != (predict_args.bond_features_path is None):
+        raise ValueError('The use of bond descriptors is different between training and prediction. If you used bond'
+                         'descriptors for training, please specify a path to new bond descriptors for prediction.')
+
+    if not for_molecule_fingerprint:
+        # If features were used during training, they must be used when predicting
+        if ((train_args.features_path is not None or train_args.features_generator is not None)
+                and predict_args.features_path is None
+                and predict_args.features_generator is None):
```

Comment:
I know this logic wasn't changed in this PR, but based on the error message, it seem like it should be changed to `((train_args.features_path is None) != (predict_args.features_path is None)) or ((train_args.features_generator is None) != (predict_args.features_generator is None))`. What do you think?

Reply:
I agree. Now it's a positive check and a negative check on both at the same time.